### PR TITLE
Jinchao update ESLint and Prettier config

### PR DESCRIPTION
**Pull request by suigunfjc**
# Description

This pull request aims to update the configuration files to ensure that the codebase aligns with the majority of ESLINT rules while making minimal changes. The goal is to maintain consistency and adhere to the recommended coding standards as closely as possible.

## Related PRS (if any):
This PR comes from [Yiyun](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/compare/Yiyun-experi_formatting) and [Nicolle](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/compare/development...nicolle_re_enable_eslint)'s previous work

For a smooth and less error-prone update process, this PR intentionally excludes changes to the .vscode/settings.json file and husky configuration. By focusing on ESlint and Prettier configs of the codebase, we aim to minimize any potential bugs and ensure a seamless transition while still implementing the necessary updates.

## Main changes explained:
Compared with previous ESLint rules, this PR:

- Includes minimum plugins: (`react/recommended`,`react-hooks/recommended`,`react/jsx-runtime`, `import/recommended`)
- Turns some rules off (`react/prop-types` and `react-hooks/exhaustive-deps`)

Compared with previous prettier rules, this PR:
- set `endOfLine` to `auto` to make windows user won't be bothered by end of line errors(CRLF/LF):

Add combined commands:

- `npm run format:check` to check if the codes follows esLint rules and matches prettier styles.
- `npm run format` to do an auto fix. (Some esLint rules still need to be addressed manually)

## How to test:
1. check into current branch
2. do `npm install`
3. Currently all component docs and major code files are included in .eslintignore and .prettierignore. Open any files to make sure there is no error/warn message from esLint or Prettier. Run `npm run format:check` to see the result:image
4. Add `#` to some files in `.eslintignore` and `.prettierignore`. Run `npm run format:check` again to see the result:image
5. Run `npm run format` to do an auto fix(Some eslint errors need a manual fix)

## Note:
Include the information the reviewers need to know.

### CODE REVIEW

**Comment by cone-7** [comment]
were all component folders added to .eslintignore file because they will be gradually enabled?

**Comment by suigunfjc** [comment]
@cone-7 Correct. Feel free to test the rule on your most familiar documents and files to determine if any further improvements can be made.

File: .eslintignore [hunk 1]
```diff
@@ -1 +1,51 @@
...
+src/registerServiceWorker.js
+src/routes.js
+src/services/**
+src/setupTests.js
+src/store.js
+src/utils/**
+src/__tests__/**
+
+
+#src/components/Admin/**
```
Comment (EvianTan) [inline .eslintignore:17]: Don't forget to uncomment this line.

Comment (suigunfjc) [reply .eslintignore]: Done

**Review by RJBornToWin — APPROVED** [review]
Since all the files are added to ignore, this wont impact the code in anyway unless triggered through npm script/command after commenting the file reference in ignore files. I see react recommended linting is followed. Suggest more reviews on this.

**Comment by RJBornToWin** [comment]
Suggest more reviews on this.

**Review by EvianTan — APPROVED** [review]

**Comment by one-community** [comment]
Still labeled as "draft", so can't merge this

**Comment by suigunfjc** [comment]
@one-community Opened

**Pull request merged**